app/src/main/java/com/example/sharewhatyoucanproject/HomeFragment.kt

```diff
@@ -0,0 +1,111 @@
+        convertbtn = view.findViewById(R.id.convertbtn)
+        locationRequest = com.google.android.gms.location.LocationRequest.create()
+        checklocation.setOnClickListener(
+            View.OnClickListener {
+                if (context?.let { it1 ->
+                    ActivityCompat.checkSelfPermission(
+                            it1,
+                            Manifest.permission.ACCESS_FINE_LOCATION
+                        )
+                } != PackageManager.PERMISSION_GRANTED && context?.let { it1 ->
+                        ActivityCompat.checkSelfPermission(
+                                it1,
+                                Manifest.permission.ACCESS_COARSE_LOCATION
+                            )
+                    } != PackageManager.PERMISSION_GRANTED
+                ) {
+                }
+                activity?.let { it1 ->
+                    LocationServices.getFusedLocationProviderClient(it1)
+                        .requestLocationUpdates(
+                            locationRequest,
+                            object : LocationCallback() {
+                                override fun onLocationResult(locationresult: LocationResult) {
+                                    super.onLocationResult(locationresult)
+                                    if (locationresult.locations != null) {
+                                        if (locationresult.locations.size > 0) {
+                                            val index: Int = locationresult.locations.size - 1
+                                            latitude =
+                                                locationresult.locations.get(index).latitude
+                                            longitute =
+                                                locationresult.locations.get(index).longitude
+                                            Toast.makeText(
+                                                context,
+                                                "latitude:- " + latitude + " " + "longitute: " + longitute,
+                                                Toast.LENGTH_SHORT
+                                            ).show()
+                                        } else {
+                                            Toast.makeText(
+                                                context,
+                                                "size is 0: " + locationresult.locations.size,
+                                                Toast.LENGTH_SHORT
+                                            ).show()
+                                        }
+                                    } else {
+                                        Toast.makeText(context, "it is null", Toast.LENGTH_SHORT).show()
+                                    }
+                                }
+                            },
+                            Looper.getMainLooper()
+                        )
+                }
+            }
```

Comment:
This is a lot.  Is there no way to use a helper function here?  Also, looks like an event listener inside an event listener, which seems odd.